### PR TITLE
Add 5 new rules

### PR DESCRIPTION
**Pull request by jianyexi**
R4009 - R4013
see , https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md

### CODE REVIEW

File: src/typescript/azure-openapi-validator/rules/DeleteOperationResponses.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from "../rule"
+export const DeleteOperationResponses: string = "DeleteOperationResponses"
+
+const jp = require("jsonpath")
```
Comment (ruowan) [inline src/typescript/azure-openapi-validator/rules/DeleteOperationResponses.ts:8]: unused jq. pls remove it.

File: src/typescript/azure-openapi-validator/rules/IntegerTypeMustHaveFormat.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  name: IntegerTypeMustHaveFormat,
+  severity: "error",
+  category: "ARMViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm | OpenApiTypes.dataplane,
+  appliesTo_JsonQuery: "$..*[?(@.type == 'integer')]",
+  *run(doc, node, path) {
+    const msg: string = `The integer type does not have a format, please add it.`
+    const formats = ["int32", "int64"]
+    if (!node.format || formats.indexOf(node.format) === -1) {
```
Comment (ruowan) [inline src/typescript/azure-openapi-validator/rules/IntegerTypeMustHaveFormat.ts:19]: please add integer type "int32" or "int64"?

File: src/typescript/azure-openapi-validator/rules/RequiredDefaultResponse.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from "../rule"
+export const RequiredDefaultResponse: string = "RequiredDefaultResponse"
+
+const jp = require("jsonpath")
```
Comment (ruowan) [inline src/typescript/azure-openapi-validator/rules/RequiredDefaultResponse.ts:8]: unused vairbale.

File: src/typescript/azure-openapi-validator/rules/RequiredSystemDataInNewApiVersions.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  id: "R4009",
+  name: RequiredSystemDataInNewApiVersions,
+  severity: "warning",
+  category: "ARMViolation",
+  mergeState: MergeStates.composed,
+  openapiType: OpenApiTypes.arm | OpenApiTypes.dataplane,
+  appliesTo_JsonQuery: "",
+  async *asyncRun(doc, node, path) {
+    if (doc.info) {
+      const apiVersion = doc.info.version
```
Comment (ruowan) [inline src/typescript/azure-openapi-validator/rules/RequiredSystemDataInNewApiVersions.ts:20]: Do all specs api-version follow this format? I think some RP doesn't follow this format.

File: src/typescript/azure-openapi-validator/rules/XmsPageableMustHaveCorrespondingResponse.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  openapiType: OpenApiTypes.arm,
+  appliesTo_JsonQuery: "$.paths..['x-ms-pageable']",
+  async *asyncRun(doc, node, path) {
+    const nextLinkValue = node.nextLinkName
+    // null is allowed
+    if (!nextLinkValue) {
+      return
+    }
+    const parentNode = nodes(doc, stringify(path.slice(0, path.length - 1)))[0].value
+    const mostSuccesskey = getMostSuccessfulResponseKey(Object.keys(parentNode.responses))
```
Comment (ruowan) [inline src/typescript/azure-openapi-validator/rules/XmsPageableMustHaveCorrespondingResponse.ts:25]: mostSuccessKey mean?

File: src/typescript/azure-openapi-validator/rules/IntegerTypeMustHaveFormat.ts (hunk 2 above)
Comment (jianyexi) [reply src/typescript/azure-openapi-validator/rules/IntegerTypeMustHaveFormat.ts:19]: not this, To add is  `"fotmat":"int32" `

File: src/typescript/azure-openapi-validator/rules/RequiredDefaultResponse.ts (hunk 3 above)
Comment (jianyexi) [reply src/typescript/azure-openapi-validator/rules/RequiredDefaultResponse.ts]: fixed

File: src/typescript/azure-openapi-validator/rules/RequiredSystemDataInNewApiVersions.ts (hunk 4 above)
Comment (jianyexi) [reply src/typescript/azure-openapi-validator/rules/RequiredSystemDataInNewApiVersions.ts:20]: all management-plane following this format

**Pull request merged**